azure/services/managedclusters/spec.go

```diff
@@ -0,0 +1,423 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package managedclusters
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2021-05-01/containerservice"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/converters"
+)
+
+// ManagedClusterSpec contains properties to create a managed cluster.
+type ManagedClusterSpec struct {
+	// Name is the name of this AKS Cluster.
+	Name string
+
+	// ResourceGroup is the name of the Azure resource group for this AKS Cluster.
+	ResourceGroup string
+
+	// NodeResourceGroupName is the name of the Azure resource group containing IaaS VMs.
```

Comment:
I see that you changed the name of `ResourceGroupName` to `ResourceGroup` while breaking out the existing `ManagedClusterSpec` into two types. Would this be a good time to consider changing the `NodeResourceGroupName` property name to something a little more accurately descriptive? For example, `InfraResourceGroupName` (or maybe better `InfraResourceGroup` to match the omission of the `Name` suffix above.

I'm wondering if having a more general name will make things less confusing in the future when folks are using this property to do things on non-node resources in the `MC_` resource group. Also (admittedly this is exceedingly speculative) if AKS were to ever implement a "control plane-only" feature (i.e., bring your own nodes) this name would then actually be misleading.

Reply:
AKS API actually refers to it as "nodeResourceGroup" https://docs.microsoft.com/en-us/rest/api/aks/managed-clusters/get#managedcluster

properties.nodeResourceGroup | string | The name of the resource group containing agent pool nodes.
-- | -- | --

Will change it to `NodeResourceGroup` for consistency.

Reply:
adhering to AKS's API makes the most sense 👍

Reply:
changed to `NodeResourceGroup`